Formula/fbthrift.rb

```diff
@@ -30,14 +30,27 @@ class Fbthrift < Formula
   uses_from_macos "flex" => :build
   uses_from_macos "zlib"
 
+  on_macos do
+    depends_on "llvm" if DevelopmentTools.clang_build_version <= 1100
+  end
+
   on_linux do
     depends_on "gcc@10"
   end
 
+  fails_with :clang do
+    build 1100
+    cause <<~EOS
+      error: 'asm goto' constructs are not supported yet
+    EOS
+  end
+
   fails_with gcc: "5" # C++ 17
   fails_with gcc: "11" # https://github.com/facebook/folly#ubuntu-lts-centos-stream-fedora
 
   def install
+    ENV.llvm_clang if OS.mac? && (DevelopmentTools.clang_build_version <= 1100)
```

Comment:
You can probably make LLVM a build dependency with
```ruby
ENV.remove "HOMEBREW_LIBRARY_PATHS", Formula["llvm"].opt_lib
```
But I guess one isn't really too inclined to uninstall `llvm` after having to build it from source...

Reply:
Since we're on the subject, I notice a few formulae like `xgboost` have _both_ incantations (`ENV.llvm_clang` and `ENV.remove`). What's the functional difference between the two?

Reply:
`ENV.llvm_clang` tells superenv to compile with LLVM Clang. The `ENV.remove` call prevents `brew` from linking opportunistically with LLVM libraries (typically `libc++`) when doing so.